### PR TITLE
(feature): Composer can be used through Docker to install dependencies

### PR DESCRIPTION
**Pull request by tahb**
* I don’t have PHP or composer installed on my machine and it was no simple task to install it manually. Brew is unable to find it and online guides are numerous and out of date eg: https://gist.github.com/tomysmile/3b37ab4a1ddd604093fe724d0a882166. While I understand whippet cannot be run in Docker yet, Composer I found could be. This takes a lot of environment specific setup out of the process of getting starting quickly with Whippet - which was my only goal and worth presenting back to the repo as an idea.

### CODE REVIEW

**Comment by tahb** [comment]
CI isn't happy that the packages are for a higher PHP version than it expects.

1 of many errors:
```
doctrine/annotations 1.7.x-dev requires php ^7.1 -> your PHP version (7.0.25) does not satisfy that requirement.
```

My packages need to be installed when on PHP 7.0.x it would seem.

**Comment by tahb** [comment]
CI passing for PHP 7, failing for 5.5 and 5.6. @robbiepaul believes that we can remove these from CI as we are >7 on all servers now. If this is true would someone else remind moving these checks please?

**Comment by tomdxw** [comment]
> I don’t have PHP or composer installed on my machine and it was no simple task to install it manually.

You don't need composer. You can just run `bin/whippet`.

`bin/whippet` falls back to vendor.phar if you haven't run `composer install`: https://github.com/dxw/whippet/blob/master/bin/whippet#L11-L13

IIRC, we added that because we wanted `whippet` on our servers but we didn't want to or we couldn't run `composer install`.

Perhaps we should update the README file to state that `composer install` is not necessary.

> My packages need to be installed when on PHP 7.0.x it would seem.

Add a `.composer/config.json` file: https://github.com/dxw/workshop-tomdxw/blob/master/dotfiles/.composer/config.json

**Comment by tahb** [comment]
@tomdxw Thanks for that. After speaking to Robbie on this too I think it's fair to say that `composer` is only required when developing this project? I've changed the readme to better reflect that. Is the rest okay?

File: readme.md [hunk 1]
```diff
@@ -23,11 +23,26 @@ During development, whippet can be used in conjunction with [wpc](https://github
...
-To install Whippet, clone this directory and install its dependencies:
+To install Whippet, clone this directory:
 
 ```
 $ git clone --recurse-submodules https://github.com/dxw/whippet.git
+```
+
+Add it to your path:
+```
+$ cp whippet/bin/whippet /usr/local/bin/whippet
```
Comment (tomdxw) [inline readme.md]: This won't work. The composer deps are complied into `vendor.phar`, not `bin/whippet`.

Try this instead: `sudo ln -s `pwd`/whippet/bin/whippet /usr/local/bin/whippet`

Comment (tahb) [reply readme.md]: Ah, the way I moved the binary out might explain my original problem that got me thinking I needed composer locally to run whippet. 👍

**Comment by mallorydxw** [comment]
I have no objections if Tom doesn't.

**Comment by tahb** [comment]
Happy to

**Pull request closed without merging**